components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -173,49 +163,25 @@ def validate_query_engine_package_compatibility(self):
 
         return self
 
+    def dump_to_primitive_dict(self):
+        d = self.model_dump()
+        d["storage_engine"] = d["storage_engine"].value
+        d["query_engine"] = d["query_engine"].value
+        return d
+
 
 class Database(BaseModel):
-    type: str = "mariadb"
-    host: str = "localhost"
-    port: int = 3306
-    name: str = "clp-db"
-    ssl_cert: Optional[str] = None
+    type: DatabaseEngine = DatabaseEngine.MARIADB
+    host: Host = "localhost"
+    port: Port = 3306
+    name: NonEmptyStr = "clp-db"
+    ssl_cert: Optional[NonEmptyStr] = None
     auto_commit: bool = False
     compress: bool = True
 
     username: Optional[str] = None
```

Comment:
i didn't change this to `Optional[NonEmptyStr]` because it's possible for a mysql db username to be an empty string (though usually it's not a good idea)